core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -288,43 +288,126 @@ def train_multi_levels(self, X, y, hyperparameters: dict, X_val=None, y_val=None
+        as_child: bool, default = True
+            If True, use the inference time of only 1 child model for bags instead of the overall inference time of the bag.
+            This is useful if the intent is to refit the models, as this will best estimate the inference time of the refit model.
+        verbose: bool, default = True
+            Whether to log the models that are removed.
+
+        Returns
+        -------
+        Returns valid subset of models that satisfy constraints.
+        """
+        if infer_limit is None or not base_model_names:
+            return base_model_names
+
+        base_model_names = base_model_names.copy()
+        num_models_og = len(base_model_names)
+        infer_limit_threshold = infer_limit * infer_limit_modifier  # Add headroom
+
+        if as_child:
+            attribute = 'predict_1_child_time'
+        else:
+            attribute = 'predict_1_time'
+
+        predict_1_time_full_set = self.get_model_attribute_full(model=base_model_names, attribute=attribute)
+
+        messages_to_log = []
+
+        base_model_names_copy = base_model_names.copy()
+        for base_model_name in base_model_names_copy:
+            predict_1_time_full = self.get_model_attribute_full(model=base_model_name, attribute=attribute)
+            if predict_1_time_full >= infer_limit_threshold:
+                predict_1_time_full_set_old = predict_1_time_full_set
+                base_model_names.remove(base_model_name)
+                predict_1_time_full_set = self.get_model_attribute_full(model=base_model_names, attribute=attribute)
+                if verbose:
+                    messages_to_log.append(f'\t{round(predict_1_time_full_set_old, 4)}s -> {round(predict_1_time_full_set, 4)}s ({base_model_name})')
+
+        score_val_dict = self.get_models_attribute_dict(attribute='val_score', models=base_model_names)
+        sorted_scores = sorted(score_val_dict.items(), key=lambda x: x[1])
+        i = 0
+        while base_model_names and (predict_1_time_full_set >= infer_limit_threshold):
+            # TODO: Incorporate score vs inference speed tradeoff in a smarter way
+            base_model_to_remove = sorted_scores[i][0]
+            predict_1_time_full_set_old = predict_1_time_full_set
+            base_model_names.remove(base_model_to_remove)
+            i += 1
+            predict_1_time_full_set = self.get_model_attribute_full(model=base_model_names, attribute=attribute)
+            if verbose:
+                messages_to_log.append(f'\t{round(predict_1_time_full_set_old, 4)}s -> {round(predict_1_time_full_set, 4)}s ({base_model_to_remove})')
+
+        if messages_to_log:
+            logger.log(20, f'Removing {len(messages_to_log)}/{num_models_og} base models to satisfy inference constraint '
+                           f'(constraint={round(infer_limit_threshold, 4)}s) ...')
+            for msg in messages_to_log:
+                logger.log(20, msg)
```

Comment:
ditto; + please add 1-liner to describe intended algorithm here (so if we have to touch the code, there's no need for deep-dive).

Reply:
What do you mean by 1-liner to describe algorithm? I have detailed method documentation detailing implementation strategy.

Reply:
added 1-liner